cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -117,34 +118,67 @@ Result<std::shared_ptr<ArrowType>> MakeArrowTimestamp(const LogicalType& logical
 Result<std::shared_ptr<ArrowType>> FromByteArray(
     const LogicalType& logical_type, const ArrowReaderProperties& reader_properties,
     const std::shared_ptr<const ::arrow::KeyValueMetadata>& metadata) {
+  auto binary_type = [&]() -> Result<std::shared_ptr<ArrowType>> {
+    const auto configured_binary_type = reader_properties.binary_type();
+    switch (configured_binary_type) {
+      case ::arrow::Type::BINARY:
+        return ::arrow::binary();
+      case ::arrow::Type::LARGE_BINARY:
+        return ::arrow::large_binary();
+      case ::arrow::Type::BINARY_VIEW:
+        return ::arrow::binary_view();
+      default:
+        return Status::TypeError("Invalid Arrow type for BYTE_ARRAY columns: ",
+                                 ::arrow::internal::ToString(configured_binary_type));
+    }
+  };
+
+  auto utf8_type = [&]() -> Result<std::shared_ptr<ArrowType>> {
```

Comment:
OK, this answers my question in `properties.h`. Now we might need some comments to explain that the property applies to string type as well.

Reply:
I agree. I'm not sure how to word this concisely, though, do you want to suggest a wording for the docstring?

Reply:
What about providing a markdown table to demonstrate the outcome of each supported value  like:

| Value \ Type | String-like  | Binary-like  |
| ------------ | ------------ | ------------ |
| BINARY       | STRING       | STRING       |
| LARGE_BINARY | LARGE_STRING | LARGE_BINARY |
| BINARY_VIEW  | STRING_VIEW  | BINARY_VIEW  |

Reply:
Hmm, I went up with some prose instead. Can you please take a look?

Reply:
Looks good, thanks!